python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -352,3 +353,47 @@ def _(
         return new_node, partition_info
 
     return ir, {ir: PartitionInfo(count=1)}  # pragma: no cover
+
+
+@lower_ir_node.register(Sink)
+def _(
+    ir: Sink, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    child, partition_info = rec(ir.children[0])
+    if partition_info[child].count > 1:
+        # Create a directory to write output files
+        Path(ir.path).mkdir(parents=True)
```

Comment:
What behavior do we want if this path already exists? Error, or continue on with `exist_ok=True`? Matching polars is probably best.

And would there be any way to delay doing this until we get to `do_evaluate`? It's a little unfortunate to have this side-effect here in the otherwise pure `lower_ir_node`.

Reply:
I think we can probably avoid side effects in `lower_ir_node`. We don't really need to create the directory here, we can probably do it in a compute task. In  `lower_ir_node`, we should probably raise if the path already exists as a file **or** a directory. I don't think Polars has stable overwrite/append options.

>Matching polars is probably best.

This might be tricky - Polars is typically writing to a single file :/